### ISSUE TITLE
sst cli doesn't run on readonly linux distros like nixOS

### ISSUE THREAD

**Issue by vincent-thomas** [sst/ion#1127]
```
$ pnpm sst deploy --stage dev --print-logs
time=2024-09-23T18:11:47.901+02:00 level=INFO msg="checking for pulumi" path=/home/vt/.config/sst/bin/pulumi
time=2024-09-23T18:11:48.002+02:00 level=INFO msg="checking for bun" path=/home/vt/.config/sst/bin/bun
time=2024-09-23T18:11:48.003+02:00 level=INFO msg="bun install"
time=2024-09-23T18:11:48.003+02:00 level=INFO msg="bun downloading" url=https://github.com/oven-sh/bun/releases//download/bun-v1.1.29/bun-linux-x64-baseline.zip
time=2024-09-23T18:11:52.192+02:00 level=INFO msg="initializing project" version=3.1.35
time=2024-09-23T18:11:52.192+02:00 level=INFO msg="esbuild building" out=/home/vt/personal/trustpass-web/.sst/platform/sst.config.1727107912192.mjs
time=2024-09-23T18:11:52.195+02:00 level=INFO msg="esbuild built" outfile=/home/vt/personal/trustpass-web/.sst/platform/sst.config.1727107912192.mjs
time=2024-09-23T18:11:52.225+02:00 level=INFO msg="evaluating config"
time=2024-09-23T18:11:52.249+02:00 level=INFO msg="config evaluated"
time=2024-09-23T18:11:52.250+02:00 level=INFO msg="checking platform"
time=2024-09-23T18:11:52.250+02:00 level=INFO msg="installing deps"
⠋  Installing providers...time=2024-09-23T18:11:52.250+02:00 level=INFO msg="getting package" name=@sst-provider/aws version=latest
⠼  Installing providers...time=2024-09-23T18:11:52.692+02:00 level=INFO msg="getting package" name=@pulumi/aws version=latest
⠦  Installing providers...time=2024-09-23T18:11:52.939+02:00 level=INFO msg="writing package.json"
time=2024-09-23T18:11:52.939+02:00 level=INFO msg="adding dependency" name=aws
time=2024-09-23T18:11:52.939+02:00 level=INFO msg="fetching deps"
time=2024-09-23T18:11:52.939+02:00 level=ERROR msg="exited with error" err="failed to run bun install Could not start dynamically linked executable: /home/vt/.config/sst/bin/bun\nNixOS cannot run dynamically linked executables intended for generic\nlinux environments out of the box. For more information, see:\nhttps://nix.dev/permalink/stub-ld\n"
✕  Could not install dependencies

$ bun --version
1.1.27
```

**Comment by thdxr** [comment]
can you update to latest and run with `NO_BUN`

**Comment by vincent-thomas** [comment]
# TLDR
This is the gist 
```sh
time=2024-09-29T22:06:52.306+02:00 level=INFO msg="waiting for command to finish" cmd="/nix/store/izpf49b74i15pcr9708s3xdwyqs4jxwl-bash-5.2p32/bin/sh -c bun run build"
NixOS cannot run dynamically linked executables intended for generic
linux environments out of the box. For more information, see:
https://nix.dev/permalink/stub-ld
```

NixOS cant run dynamically linked binaries

```sh
22:06:20 vt-pc :: repro λ NO_BUN="1" bunx sst deploy --print-logs
time=2024-09-29T22:06:32.372+02:00 level=INFO msg="checking for pulumi" path=/[REDACTED]/.config/sst/bin/pulumi
time=2024-09-29T22:06:32.460+02:00 level=INFO msg="initializing project" version=3.1.49
time=2024-09-29T22:06:32.461+02:00 level=INFO msg="esbuild building" out=[REDACTED]/.sst/platform/sst.config.1727640392461.mjs
time=2024-09-29T22:06:32.463+02:00 level=INFO msg="esbuild built" outfile=[REDACTED]/.sst/platform/sst.config.1727640392461.mjs
time=2024-09-29T22:06:33.071+02:00 level=INFO msg="evaluating config"
time=2024-09-29T22:06:34.899+02:00 level=INFO msg="config evaluated"
time=2024-09-29T22:06:34.900+02:00 level=INFO msg="checking platform"
time=2024-09-29T22:06:34.900+02:00 level=INFO msg="loading home"
time=2024-09-29T22:06:35.323+02:00 level=INFO msg="aws credentials found" region=[REDACTED]
time=2024-09-29T22:06:35.323+02:00 level=INFO msg="fetching bootstrap"
time=2024-09-29T22:06:35.458+02:00 level=INFO msg="found existing bootstrap" data="{[REDACTED]}"
time=2024-09-29T22:06:35.458+02:00 level=INFO msg="loaded config" [REDACTED]
time=2024-09-29T22:06:35.458+02:00 level=INFO msg="initializing ui" isTTY=true
time=2024-09-29T22:06:35.458+02:00 level=INFO msg="server port assigned" port=13557
time=2024-09-29T22:06:35.458+02:00 level=INFO msg="running stack command" cmd=deploy
time=2024-09-29T22:06:35.458+02:00 level=INFO msg=publishing type=*project.StackCommandEvent
time=2024-09-29T22:06:35.458+02:00 level=INFO msg="INFO locking app=[REDACTED]
time=2024-09-29T22:06:35.458+02:00 level=INFO msg="INFO getting data key=lock app=[REDACTED]
SST 3.1.49  ready!

➜  App:        repro
time=2024-09-29T22:06:35.459+02:00 level=INFO msg=server addr=0.0.0.0:13557
   Stage:      vt

~  Deploy

time=2024-09-29T22:06:35.564+02:00 level=INFO msg="INFO putting data key=lock app=[REDACTED] stage=[REDACTED]"
time=2024-09-29T22:06:35.676+02:00 level=INFO msg="INFO pulling state app=[REDACTED] stage=[REDACTED] out=[REDACTED]
time=2024-09-29T22:06:35.789+02:00 level=INFO msg="INFO getting passphrase app=[REDACTED] stage=[REDACTED]"
time=2024-09-29T22:06:35.936+02:00 level=INFO msg="INFO getting data key=secret app=[REDACTED] stage=_fallback"
time=2024-09-29T22:06:35.936+02:00 level=INFO msg="INFO getting data key=secret app=[REDACTED] stage=[REDACTED]"
time=2024-09-29T22:06:36.127+02:00 level=INFO msg="built workspace"
time=2024-09-29T22:06:45.011+02:00 level=INFO msg="built stack"
time=2024-09-29T22:06:46.154+02:00 level=INFO msg=publishing type=*project.CompleteEvent
time=2024-09-29T22:06:46.154+02:00 level=INFO msg="got previous deployment"
time=2024-09-29T22:06:46.154+02:00 level=INFO msg="esbuild building" out=[REDACTED]
time=2024-09-29T22:06:46.176+02:00 level=INFO msg="esbuild built" outfile=[REDACTED]
time=2024-09-29T22:06:46.360+02:00 level=INFO msg=publishing type=*project.BuildSuccessEvent
time=2024-09-29T22:06:46.360+02:00 level=INFO msg="tracked files"
time=2024-09-29T22:06:46.450+02:00 level=INFO msg="built config"
time=2024-09-29T22:06:46.450+02:00 level=INFO msg="running stack command" cmd=deploy
time=2024-09-29T22:06:47.582+02:00 level=INFO msg=publishing type=*apitype.PreludeEvent
time=2024-09-29T22:06:50.004+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:50.005+02:00 level=INFO msg=publishing type=*apitype.ResourcePreEvent
time=2024-09-29T22:06:50.008+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:50.016+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:50.021+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:50.021+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:50.022+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:50.673+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:50.674+02:00 level=INFO msg=publishing type=*apitype.ResourcePreEvent
time=2024-09-29T22:06:50.674+02:00 level=INFO msg=publishing type=*apitype.ResourcePreEvent
time=2024-09-29T22:06:50.678+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:50.692+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:50.700+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:50.701+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:50.702+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:50.703+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:50.707+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:50.738+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:50.775+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:50.776+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:50.777+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:50.777+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:50.778+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:50.778+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:50.779+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:50.781+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:50.782+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:50.782+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:50.783+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:50.783+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:50.784+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:50.784+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:50.785+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:50.786+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:50.786+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:50.787+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:50.787+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:50.787+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:50.788+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:50.789+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:50.789+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:50.792+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:50.793+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:50.793+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:50.798+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:50.799+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:50.799+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:51.341+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:51.342+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:51.343+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:51.547+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:52.023+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:52.025+02:00 level=INFO msg=publishing type=*apitype.ResourcePreEvent
time=2024-09-29T22:06:52.025+02:00 level=INFO msg=publishing type=*apitype.ResourcePreEvent
time=2024-09-29T22:06:52.025+02:00 level=INFO msg=publishing type=*apitype.ResOutputsEvent
time=2024-09-29T22:06:52.106+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:52.133+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:52.143+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:52.147+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:52.154+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:52.160+02:00 level=INFO msg=publishing type=*apitype.ResOutputsEvent
time=2024-09-29T22:06:52.160+02:00 level=INFO msg=publishing type=*apitype.ResourcePreEvent
time=2024-09-29T22:06:52.160+02:00 level=INFO msg=publishing type=*apitype.ResOutputsEvent
time=2024-09-29T22:06:52.166+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:52.170+02:00 level=INFO msg=publishing type=*apitype.ResourcePreEvent
⠧  Creating    MyWeb sst:aws:Remix → MyWebBuild sst:Run
⠧  Creating    MyWeb sst:aws:Remix → MyWebBuild sst:Run
⠧  Creating    MyWeb sst:aws:Remix → MyWebBuild sst:Run
⠧  Creating    MyWeb sst:aws:Remix → MyWebBuild sst:Run
time=2024-09-29T22:06:52.191+02:00 level=INFO msg=publishing type=*apitype.ResourcePreEvent
⠇  Creating    MyWeb sst:aws:Remix → MyWebBuild sst:Run
⠇  Creating    MyWeb sst:aws:Remix → MyWebBuild sst:Run
⠇  Creating    MyWeb sst:aws:Remix → MyWebAssetsId random:index:RandomId
time=2024-09-29T22:06:52.303+02:00 level=INFO msg="rpc request" method=POST url=/rpc
Could not start dynamically linked executable: bun
time=2024-09-29T22:06:52.306+02:00 level=INFO msg=publishing type=*common.StdoutEvent
time=2024-09-29T22:06:52.306+02:00 level=INFO msg=publishing type=*common.StdoutEvent
time=2024-09-29T22:06:52.306+02:00 level=INFO msg="waiting for command to finish" cmd="/nix/store/izpf49b74i15pcr9708s3xdwyqs4jxwl-bash-5.2p32/bin/sh -c bun run build"
NixOS cannot run dynamically linked executables intended for generic
linux environments out of the box. For more information, see:
https://nix.dev/permalink/stub-ld
⠇  Creating    MyWeb sst:aws:Remix → MyWebBuild sst:Run
|  Error       MyWeb sst:aws:Remix → MyWebBuild sst:Run
|  command exited with code 127
⠇  Creating    MyWeb sst:aws:Remix → MyWebAssetsId random:index:RandomId
⠇  Creating    MyWeb sst:aws:Remix → MyWebAssetsId random:index:RandomId
⠋  Creating    MyWeb sst:aws:Remix → MyWebAssetsId random:index:RandomId
⠋  Creating    MyWeb sst:aws:Remix → MyWebAssetsId random:index:RandomId
⠏  Creating    MyWeb sst:aws:Remix → MyWebAssetsId random:index:RandomId
|  Created     MyWeb sst:aws:Remix → MyWebAssetsId random:index:RandomId (4.1s)
time=2024-09-29T22:06:57.862+02:00 level=INFO msg=publishing type=*apitype.ResOutputsEvent
time=2024-09-29T22:06:57.862+02:00 level=INFO msg=publishing type=*apitype.SummaryEvent
time=2024-09-29T22:06:57.862+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:57.862+02:00 level=INFO msg=publishing type=*apitype.DiagnosticEvent
time=2024-09-29T22:06:57.927+02:00 level=INFO msg=publishing type=*apitype.CancelEvent
time=2024-09-29T22:06:58.368+02:00 level=INFO msg="done running stack command"
time=2024-09-29T22:06:58.368+02:00 level=ERROR msg="stack run failed" error="failed to run update: exit status 255\ncode: 255\nstdout: Updating (vt):\n\n@ updating.......\n +  pulumi-nodejs:dynamic:Resource MyWebBuild.sst.Run creating (0s) \n +  random:index:RandomId MyWebAssetsId creating (0s) \n +  pulumi-nodejs:dynamic:Resource MyWebBuild.sst.Run creating (0s) error: command exited with code 127\n +  pulumi-nodejs:dynamic:Resource MyWebBuild.sst.Run **creating failed** error: command exited with code 127\n@ updating.......\n +  random:index:RandomId MyWebAssetsId created (4s) \n@ updating.....\n    pulumi:pulumi:Stack repro-vt running error: update failed\n    pulumi:pulumi:Stack repro-vt **failed** 1 error\nDiagnostics:\n  pulumi-nodejs:dynamic:Resource (MyWebBuild.sst.Run):\n    error: command exited with code 127\n\n  pulumi:pulumi:Stack (repro-vt):\n    error: update failed\n\nResources:\n    + 1 created\n    5 unchanged\n\nDuration: 11s\n\n\nstderr: \n"
time=2024-09-29T22:06:58.368+02:00 level=INFO msg="INFO putting summary app=[REDACTED] stage=[REDACTED]"
time=2024-09-29T22:06:58.368+02:00 level=INFO msg="INFO putting data key=summary app=[REDACTED] stage=[REDACTED]/av9z5oasvpbts93lob6pqano"
time=2024-09-29T22:06:58.487+02:00 level=INFO msg="parsing state"
time=2024-09-29T22:06:58.719+02:00 level=INFO msg="generating types" root=[REDACTED]
time=2024-09-29T22:06:58.805+02:00 level=INFO msg=publishing type=*project.CompleteEvent
time=2024-09-29T22:06:58.805+02:00 level=INFO msg="done parsing state"

✕  Failed
   MyWeb sst:aws:Remix → MyWebBuild sst:Run
   command exited with code 127


time=2024-09-29T22:06:58.805+02:00 level=INFO msg="INFO pushing state app=repro stage=[REDACTED] from=[REDACTED]/.sst/.pulumi/stacks/repro/*.json"
time=2024-09-29T22:06:58.988+02:00 level=INFO msg="INFO unlocking app=[REDACTED] stage=[REDACTED]"
time=2024-09-29T22:06:59.096+02:00 level=INFO msg="shutting down server"
time=2024-09-29T22:06:59.096+02:00 level=INFO msg="server done"
time=2024-09-29T22:06:59.096+02:00 level=ERROR msg="exited with error" err="stack run had errors"

```

**Comment by vincent-thomas** [comment]
the binary `sst` is forcing to run its own downloaded version of bun. But my installed bun, is working correctly.
